scheduleapi.py

```diff
@@ -0,0 +1,73 @@
+import json
+from flask import request
+from flask_restful import Resource, abort
+
+from validations import validate_OId
+from schedule import Schedule
+
+def read_args(request):
+    return json.loads(request.data.decode('utf-8'))
+
+def get_or_abort(s_id):
+    res = Schedule.get_by_id(s_id)
+    if res is None:
+        abort(404, message='No schedule exists with id {}'.format(s_id))
+    return res
+
+class ScheduleAPI(Resource):
+    def get(self, s_id):
+        if not validate_OId(s_id):
+            abort(400, message='Invalid id')
+
+        sched = get_or_abort(s_id)
+        return vars(sched)
+
+    def put(self, s_id):
+        if not validate_OId(s_id):
+            abort(400, message='Invalid id')
+
+        data = read_args(request)
+        sched = get_or_abort(s_id)
+
+        if 'event_id' not in data or not validate_OId(data['event_id']):
+            abort(404, message='"event_id" is missing or invalid')
+
+        try:
+            res = sched.add_event(data['event_id'])
+        except Exception as e:
+            abort(500, message='Error addinge event to schedule: {}'.format(e))
+
+        return vars(res), 201
+
+    def delete(self, s_id):
+        if not validate_OId(s_id):
+            abort(400, message='Invalid id')
+
+        data = read_args(request)
+        sched = get_or_abort(s_id)
```

Comment:
you have way too much logic going on here, just send the delete request ... have the model call the dao delete.. and have the dao handle the delete request .. the DAO should return an acknowledgement or a failure .. that is how you will handle the `404` vs `204`